src/twisted/enterprise/adbapi.py

```diff
@@ -463,18 +463,44 @@ def _close(self, conn):
     def _runInteraction(self, interaction, *args, **kw):
         conn = self.connectionFactory(self)
         trans = self.transactionFactory(self, conn)
-        try:
-            result = interaction(trans, *args, **kw)
-            trans.close()
-            conn.commit()
-            return result
-        except:
-            excType, excValue, excTraceback = sys.exc_info()
+        if self.reconnect and self.dbapiName == 'MySQLdb':
+            from MySQLdb import OperationalError
```

Comment:
As you noted in the PR's discussion, it would be effectively impossible to maintain a list of all Python MySQL drivers and what exceptions each raises when a connection has ended but can be re-established.  That's the biggest problem with this approach :frowning_face: